### PR TITLE
Some clean ups in the FieldError code and its tests.

### PR DESCRIPTION
**Pull request by vagababov**
- rename FieldError.getNormalizedError to be normalized:
  - Go kind of discourages GetA() vs A()
  - it's an error, so normalizing itself is also an error, so
    normalizedError() is a tautology at this point
- flatten was doing a[x]=fmt.Sprintf("%s%s", a[x],y), which is a very
  expensive substitution for a[x]+=y
- some minor test cleanups (fix strings, streamline `got, want:=`
  pattern, etc.

### CODE REVIEW

File: apis/field_error.go [hunk 1]
```diff
@@ -133,34 +133,36 @@ func (fe *FieldError) isEmpty() bool {
...
 	if fe == nil {
-		return []FieldError(nil)
+		return []*FieldError(nil)
 	}
-	var errors []FieldError
-	// if this FieldError is a leaf,
+
+	// Allocate errors with at least as many objects as we'll get on the first pass.
+	errors := make([]*FieldError, 0, len(fe.errors)+1)
+	// If this FieldError is a leaf,
```
Comment (vaikas-google) [inline apis/field_error.go]: don't leave me hanging, what happens?

File: apis/field_error.go [hunk 2]
```diff
@@ -235,27 +237,26 @@ func containsString(slice []string, s string) bool {
 // FiledErrors. FieldErrors have their Paths combined (and de-duped) if their
```
Comment (vaikas-google) [inline apis/field_error.go]: while here, FiledErrors -> FieldErrors?

File: apis/field_error.go (hunk 1 above)
Comment (vagababov) [reply apis/field_error.go]: well :) it was like this. I'll change to ...

File: apis/field_error.go (hunk 2 above)
Comment (vagababov) [reply apis/field_error.go]: Eh, didn't notice this one.
Unforts, the PR automerged, so I'll send a new one.

**Pull request merged**